### DISCUSSION TITLE
How to show virtual keyboard with input focus

### DISCUSSION THREAD

**Discussion by xingyun86** [ocornut/imgui#3968] [Old/Archived discussions]
I write a virtual keyboard. But How I can use it with input?
I want to know input get focus event or global event focus?

**Comment by rokups** [comment]
Dear ImGui knows nothing about keyboards. You are responsible for providing library with the text that was typed in. Please consult documentation of your platform.

**Reply by xingyun86** [reply]
Not hard keyboard, Virtual keyboard with yourself make it ui.

**Comment by ocornut** [comment]
xingyun86, it is difficult to understand precisely what you are asking but you can feed inputs to ImGui using ImGui::GetIO().AddInputCharacter().

**Reply by xingyun86** [reply]
I only want to make a virtual keyboard show down to my edit control get focus and kill focus.But I now don't know how to get or kill focus event.

**Reply by PathogenDavid** [reply]
Is ImGuiIO::WantTextInput what you're looking for?

**Reply by xingyun86** [reply]
> Is ImGuiIO::WantTextInput what you're looking for?

Thanks very much. It works. I will try it.

**Reply by xingyun86** [reply]
> Is ImGuiIO::WantTextInput what you're looking for?

I test it. It control virtual keyboard very good. But I found textinput lose input state. When I use virtual keyboard. Could you give me some support?

**Reply by xingyun86** [reply]
I solved it with other method。Thanks for all of you。

**Comment by xingyun86** [comment]
How can I close this issue ?

**Reply by ocornut** [reply]
You cannot "close" Discussions created in "General"
One of the many reasons why I suggested stopping to use GitHub Discussions.
